db/src/db.rs

```diff
@@ -140,6 +108,41 @@ impl RocksDB {
         Self::open_with_check(&config, columns).unwrap_or_else(|err| panic!("{}", err))
     }
 
+    /// Set appropriate parameters for bulk loading.
+    pub fn prepare_for_bulk_load_open<P: AsRef<Path>>(
+        path: P,
+        columns: u32,
+    ) -> Result<Option<Self>> {
+        let mut opts = Options::default();
+
+        opts.create_missing_column_families(true);
+        opts.set_prepare_for_bulk_load();
+
+        let cfnames: Vec<_> = (0..columns).map(|c| c.to_string()).collect();
+        let cf_options: Vec<&str> = cfnames.iter().map(|n| n as &str).collect();
+
+        OptimisticTransactionDB::open_cf(&opts, path, &cf_options).map_or_else(
+            |err| {
+                let err_str = err.as_ref();
+                if err_str.starts_with("Invalid argument:")
+                    && err_str.ends_with("does not exist (create_if_missing is false)")
+                {
+                    Ok(None)
+                } else {
+                    Err(internal_error(format!(
+                        "failed to open the database: {}",
```

Comment:
Is it better to check `err.as_ref().starts_with("Corruption:")` and tell users to try the `db-repair` sub command?

Reply:
That sounds better, I will revisit this later.